src/App.js

```diff
@@ -1,15 +1,49 @@
+import React from 'react';
+import { useState } from 'react';
 import './App.css';
 
 const App = () => {
 
+  const [count, setCount] = useState(0);
+  const [counterText, setCounterText] = useState('');
+
+  const addCounter = () => {
+    setCount(prevCount =>  prevCount  + 1)
+  }
+
+  const subtractCounter = () => {
+    setCount((prevCount) => prevCount - 1)
+  }
+
+  const SetCountText = (event) => {
+    setCounterText(event.target.value)
+  }
+
+
   return(
     <main>
         <h1>Counter App</h1>
-        <input type='text' class ="counter-text" placeholder='Enter name for your counter' />
-        <div class ="counter-count">{0}</div>
+        <input 
+          type='text' 
+          class ="counter-text" 
+          placeholder='Enter name for your counter' 
+          value={counterText} 
+          onChange={event => SetCountText(event)} 
```

Comment:
you can pass the function itself instead of calling it
similar to what you have done below